### PR TITLE
Add output time interval

### PR DESCRIPTION
**Pull request by mivaia**
# Description of the problem

- Only needed the last 30% of my simulations and they are quite heavy.

# Description of the solution

- Added a new parameter in the parameter file -> output_time_interval. If the user specifies an output time interval, the .vtu files are only written if the simulation time is within that time window. This has been implemented for output control = iteration and output control = time.

# How Has This Been Tested?

- new simulation_control_06.cc test. Makes sure information is written at the appropriated time steps.

# Documentation

- Modified the simulation_control.rst file to consider the new parameter.

### CODE REVIEW

File: doc/source/parameters/cfd/simulation_control.rst [hunk 1]
```diff
@@ -152,6 +155,8 @@ This subsection contains the general information of the simulation, including th
 
 * ``output time``: controls the time when the ``.pvd`` / ``.vtu`` results are written. This parameter is only used when ``set output control = time``.
 
+* ``output time interval``: Only writes the ``.pvd`` / ``.vtu`` files when the simulation time is within ``output time interval``. Default values are 0s and 1.7e308s.
```
Comment (blaisb) [inline doc/source/parameters/cfd/simulation_control.rst:158]:
```suggestion
* ``output time interval``: Only writes the ``.pvd`` / ``.vtu`` files when the simulation time is within the closed interval defined by the ``output time interval``. Default values are 0s and 1.7e308s.
```

File: tests/core/simulation_control_06.cc [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 3.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+
+*
+* Author: Bruno Blais, Polytechnique Montreal, 2020-
```
Comment (blaisb) [inline tests/core/simulation_control_06.cc:17]:
```suggestion

```

File: tests/core/simulation_control_06.cc [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+*/
+
+/**
+ * @brief This test checks that the steady-state simulation control stops
+ * at the correct moment and behaves in a correct manner
+ */
```
Comment (blaisb) [inline tests/core/simulation_control_06.cc:23]: Are you sure about this comment? I don't think this is what this test is checking

**Review by blaisb — CHANGES_REQUESTED** [review]
Very minor comments
Can you add a CHANGELOG entry?
Thanks :)!

File: include/core/simulation_control.h [hunk 4]
```diff
@@ -100,6 +100,11 @@ class SimulationControl
   // by the time
   double output_time_frequency;
 
+  // Output window
+  // Output file will be generated if current step number is in the output time
```
Comment (oguevremont) [inline include/core/simulation_control.h:104]: The phrasing is a bit confusing. I'd say that this adds another condition for the output (in addition to the step respecting the output frequency parameter).

**Review by mivaia — COMMENTED** [review]
Thank you for your comments! All of them have been adressed :)

**Review by oguevremont — APPROVED** [review]

**Pull request merged**